django/db/models/deletion.py

```diff
@@ -256,6 +281,26 @@ def sort(self):
                                 for model in sorted_models)
 
     def delete(self):
+        # see if we can delete restricted objects, fail if any remain
```

Comment:
This could be described in more detail. Fore example, by explaining what restricted objects are.